### PR TITLE
Allow to pass custom activation function in `policy_kwargs`

### LINKED ISSUE #37: [Feature Request] Passing custom activation functon in policy_kwargs
### 🚀 Feature

Possibility to pass a flax (from the [`flax.linen.activation`](https://flax.readthedocs.io/en/latest/api_reference/flax.linen/activation_functions.html) module) activation function when creating a `sbx` model, through the `policy_kwargs` argument.

### Motivation

In the current implementation of `sbx`, users are unable to pass custom activation functions when creating a model. This limitation restricts flexibility and may not suit all users' needs.

### Pitch

Example:

```python
policy_kwargs = dict(activation_fn=my_custom_activation_fn, net_arch=dict(pi=[64, 64], qf=[64, 64]))

model = TD3("MlpPolicy",
                       env,
                      policy_kwargs=policy_kwargs,
                      verbose=1)
```

### Idea on how to implement it 

Add attribute `activation_fn` to the underlying classes that are composing the policy (like in Critic and Actor in [`t3d/policy.py`](https://github.com/araffin/sbx/blob/db6120bdd4c630b7c141fe091e5d469626aa3aa4/sbx/td3/policies.py))

### PR DESCRIPTION
**Pull request by paolodelia99**
Added the possibility to pass a custom activation function through the `policy_kwargs` argument when creating the following models: `TD3`, `SAC`, `DDPG` and `DQN` (like in sablebaseline3). 

## Description

Taking inspiration from `stablebaseline`, I've put the common code of the critic under the `sbx\common\policy.py` file since the critic code both in the `sac` module and in `td3` module was the same. 

Minor changes to `.gitignore` and `Makefile` have been made. 

## Motivation and Context

- [x] I have raised an issue to propose this change (#37 )

closes #37 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)
- [ ] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,6 +10,8 @@
 .coverage
 .coverage.*
 __pycache__/
+**/.mypy_cache/
```
Comment (araffin) [inline .gitignore:13]: this should not be needed, `.ruff_cache/` and `.mypy_cache/` come with a `.gitignore` inside

File: tests/test_run.py [hunk 2]
```diff
@@ -72,6 +73,16 @@ def test_sac_td3(model_class) -> None:
     model.learn(110)
 
 
+@pytest.mark.parametrize("model_class", [SAC, TD3, DDPG])
+def test_sac_td3_policy_kwargs(model_class) -> None:
+    policy_kwargs = dict(activation_fn=nn.leaky_relu, net_arch=dict(pi=[64, 64], qf=[64, 64]))
```
Comment (araffin) [inline tests/test_run.py:78]:
```suggestion
    policy_kwargs = dict(activation_fn=nn.leaky_relu, net_arch=dict(pi=[8], qf=[8]))
```

to run the tests as fast as possible

File: tests/test_run.py [hunk 3]
```diff
@@ -95,6 +106,20 @@ def test_dqn() -> None:
     model.learn(128)
 
 
+def test_dqn_policy_kwargs() -> None:
+    policy_kwargs = dict(activation_fn=nn.leaky_relu, net_arch=[128, 128])
```
Comment (araffin) [inline tests/test_run.py:110]:
```suggestion
    policy_kwargs = dict(activation_fn=nn.leaky_relu, net_arch=[8])
```

**Review by araffin — COMMENTED** [review]
Overall LGTM, thanks for the PR =)

quick question: why not PPO too?

**Comment by paolodelia99** [comment]
> quick question: why not PPO too?

You can already pass the activation function in PPO, I just kept things like in `stablebaseline3`.

**Review by araffin — APPROVED** [review]
LGTM, thanks =)

**Pull request merged**